packages/vite/src/node/plugins/worker.ts

```diff
@@ -6,9 +6,55 @@ import type Rollup from 'rollup'
 import { ENV_PUBLIC_PATH } from '../constants'
 import path from 'path'
 import { onRollupWarning } from '../build'
+import type { EmittedFile } from 'rollup'
 
 const WorkerFileId = 'worker_file'
 
+// save worker bundle emitted files avoid overwrites the same file.
+// <chunk_filename, hash>
+const workerEmittedAssets = new Map<string, string>()
+const workerEmittedChunk = new Map<string, string>()
+
+// worker bundle don't deps on any more worker runtime info an id only had an result.
+// save worker bundled file id to avoid repeated execution of bundles
+// <filename, hash>
+const workerBundled = new Map<string, string>()
+
+// nested worker bundle context don't had file what emitted by outside bundle
+// save the hash to id to rewrite truth id.
+// <hash, id>
+const workerEmittedFile = new Map<string, string>()
```

Comment:
We need these caches to depend on the `ResolvedConfig`, see for example: https://github.com/vitejs/vite/blob/90df0bb7c610834f875dbfdb8dd3452f44f15627/packages/vite/src/node/plugins/asset.ts#L19

You could create individual `WeakMap`s but maybe it is a good idea to have a single `WeakMap<ResolvedConfig, WorkerCache>` since you will need to reset all the maps on the `buildStart` hook so build watch works correctly (grep Vite's code for `buildStart` to see other caches being reset)
You could then export a function instead of one of the Maps to access what is needed from the other plugin

Reply:
Thank you for your answer, let me know more about the practice of vite